doc/specs/stdlib_linalg.md

```diff
@@ -425,3 +436,43 @@ Specifically, upper Hessenberg matrices satisfy `a_ij = 0` when `j < i-1`, and l
 ```fortran
 {!example/linalg/example_is_hessenberg.f90!}
 ```
+
+## `solve` - Solve a-linear-matrix-equation
+
+### Status
+
+Experimental
+
+### Description
+
+Solve a linear matrix equation, or system of linear scalar equations.
+
+Note: `solve` supports only single and double precision floating point types.
+
+### Class
+
+Impure function.
```

Comment:
For such procedures, I would prefera subroutine over a function, especially for efficienct with large vectors/matrices.

Reply:
Subroutines are usually faster than functions, functions are usually easier to use but may cause performance degradation. If you need high performance, you should use subroutines. If you need simpler code, you should use a function.

Intuitively, solving a system of `AX=B` linear equations, the normal form `X=solve(A, B)` will naturally be understood by the average user and will be surprised by `call solve(A, B)`. My personal recommendation is to think of `solve` as a function dedicated to solving systems of linear equations, rather than for performance, which can be done with `_gesv` routines.

- (Not recommended) Perhaps we could provide a `solve_` **subroutine** that implements the in-place version of `solve`, like this:
```fortran
subroutine solve_(a, b)
    real, intent(inout) :: a(:, :), b(:, :)
    integer :: ipiv(size(a, 1)), info
    call sgesv(size(a, 1), size(b, 2), a, size(a, 1), ipiv, b, size(b, 1), info)
end subroutine solve_
```

- (Recommended) Or for performance-sensitive users, it is recommended to just use the low-level interface `_gesv` for programming by borrowing the implementation of `solve`?